contk/metric/metric.py

```diff
@@ -293,11 +297,11 @@ def forward(self, data):
 		post = data[self.post_key]
 		resp = data[self.resp_key]
 		gen = data[self.gen_key]
-		if post.shape[0] != resp.shape[0] or resp.shape[0] != gen.shape[0]:
+		if len(post) != len(resp) or len(resp) != len(gen):
 			raise ValueError("Batch num is not matched.")
-		for i in range(post.shape[0]):
-			self.post_list.append(self.dataloader.index_to_sen(post[i, 1:]))
-			self.resp_list.append(self.dataloader.index_to_sen(resp[i, 1:]))
+		for i, _ in enumerate(post):
```

Comment:
```_``` means unused variable, if you want to use it, give it a good name that others can understand it.